Library/Homebrew/cmd/bottle.rb

```diff
@@ -130,7 +130,9 @@ def bottle_formula f
     else
       ohai "Determining #{f.name} bottle revision..."
       versions = FormulaVersions.new(f)
-      max = versions.bottle_version_map("origin/master")[f.pkg_version].max
+      bottle_revisions = versions.bottle_version_map("origin/master")[f.pkg_version]
+      pkg_reset = bottle_revisions[-2].zero? && !bottle_revisions[-1].zero?
+      max = pkg_reset ? bottle_revisions[0..-2].max : bottle_revisions.max
```

Comment:
I'm not sure I quite understand this code but I do like the approach. Could you make it a bit more verbose?